cnd/src/swap_protocols/rfc003/bob/mod.rs

```diff
@@ -15,10 +18,23 @@ pub struct State<AL, BL, AA, BA, AH, BH, AI, BI, AT, BT> {
     pub beta_ledger_state: LedgerState<BA, BH, BT>,
     #[derivative(Debug = "ignore")]
     pub secret_source: Arc<dyn DeriveIdentities>,
-    pub failed: bool, // Gets set on any error during the execution of a swap.
 }
 
 impl<AL, BL, AA, BA, AH, BH, AI, BI, AT, BT> State<AL, BL, AA, BA, AH, BH, AI, BI, AT, BT> {
+    pub fn new(
+        swap_communication: SwapCommunication<AL, BL, AA, BA, AI, BI>,
+        alpha_ledger_state: LedgerState<AA, AH, AT>,
+        beta_ledger_state: LedgerState<BA, BH, BT>,
+        secret_source: SwapSeed,
+    ) -> Self {
+        Self {
+            swap_communication,
+            alpha_ledger_state,
+            beta_ledger_state,
+            secret_source: Arc::new(secret_source),
+        }
+    }
+
     pub fn proposed(
```

Comment:
Do we still need those functions?

Reply:
Nice catch! `request` is still in use but the others can go.